### PR TITLE
Holy grail: end-to-end pipeline for evaluation of Nixpkgs

### LINKED ISSUE #8: Ingestion of evaluation results of any supported channel
The security tracker acts on supported channels, and we need to ingest an evaluation of all nixpkgs for any given supported channel at any point in time.

The tracker should subscribe to channel bumps (open problem), see how https://git.qyliss.net/pr-tracker detects them and how https://git.eno.space/label-tracker.git/ tracks them.

### Proposal of implementation

Every time a channel bump, repull the repository, extract a worktree of that channel (or git clone via the reference for fast checkout), run `nix-eval-jobs` on that commit sha and collect the result and archive it as JSON with meta results (!!!).

Run this as a background job or a cron job that can easily be managed by infrastructure people or administrators to perform any maintenance task like cancelling evaluations, restarting evaluations, configuring the number of concurrent evaluations, etc.

### Ideas for the future

Expose this data of evaluations publicly and let people access it directly, it's useful in general.

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Fixes #8.

### CODE REVIEW

File: src/website/shared/listeners/nix_evaluation.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+        "Attribute '%s' has been ingested under Nix derivation ID '%d' (drv path: '%s') in %f seconds",
+        partially_evaluated_result.attr,
+        nix_derivation.pk,
+        nix_derivation.derivation_path,
+        elapsed,
+    )
+    return nix_derivation
+
+
+async def evaluation_entrypoint(avg_eval_time: float, evaluation: NixEvaluation):
```
Comment (Mic92) [inline src/website/shared/listeners/nix_evaluation.py:93]: -> None

File: src/website/shared/listeners/nix_evaluation.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+            "Failed to run the `nix-eval-job` on revision '%s', marking job as crashed...",
+            evaluation.commit_sha1,
+        )
+        await NixEvaluation.objects.filter(id=evaluation.pk).aupdate(
+            state=NixEvaluation.EvaluationState.CRASHED
+        )
+
+
+@pgpubsub.post_insert_listener(NixEvaluationChannel)
+def run_evaluation_job(old: NixEvaluation, new: NixEvaluation):
```
Comment (Mic92) [inline src/website/shared/listeners/nix_evaluation.py:178]: -> None

File: src/website/shared/evaluation.py [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+@dataclass
+class EvaluatedAttribute(JSONWizard):
+    """
+    This is a totally evaluated attribute.
+    """
+
+    attr: str
+    attrPath: list[str]
+    name: str
+    drvPath: str
```
Comment (Mic92) [inline src/website/shared/evaluation.py:81]: drv_path

Comment (RaitoBezarius) [reply src/website/shared/evaluation.py:81]: yeah all of this needs rebase over your formatting changes

Comment (Mic92) [reply src/website/shared/evaluation.py:81]: Maybe smaller PRs would help to avoid this?

Comment (Mic92) [reply src/website/shared/evaluation.py:81]: https://github.com/tummychow/git-absorb should help

Comment (RaitoBezarius) [reply src/website/shared/evaluation.py:86]: Right but @Mic92 your reformatting actually broke I believe things. So I am glad I didn't do smaller PRs :P

File: src/website/shared/channels.py [hunk 4]
```diff
@@ -19,6 +19,7 @@ class NixChannelChannel(TriggerChannel):
 @dataclass
 class NixEvaluationChannel(TriggerChannel):
     model = NixEvaluation
+    lock_notifications = True
```
Comment (fricklerhandwerk) [inline src/website/shared/channels.py:22]: What's this?

Comment (RaitoBezarius) [reply src/website/shared/channels.py:22]: I will add a comment but this enables processing of notifications exactly once except of potentially random workers grab the notification and work through it.

**Comment by RaitoBezarius** [comment]
This is now ready for merge and reviews. This piece has been quite a thing, I rewrote the damn thing 4 times because the performance was too low. Now, it is bounded by nix-eval-jobs performance, so I expect the ingestion not to be a problem anymore.

I don't advise using the manual ingestion for now, this will probably break things up and what not, I will try to create fixtures for everyone to use in replacement to that.

**Pull request merged**